### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by Aqurds**
The following features were added,

**Create a method called current_user in your ApplicationController to make current user data available to all controllers.**
- > It will return the first user from the database.
- > Don't worry about proper authentication at this time. You will add it in the upcoming projects.

**Create forms to perform the following functions:**
- > Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
- > Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).

**Allow Users to add likes to Posts.**

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,10 +1,55 @@
...
+    # from the form (permitted by post_params).
+    # It sets the initial values for comments_counter and likes_counter
+    # (assuming these are counters for comments and likes on the post).
+    # If the post is successfully saved, it redirects to the show page for the newly created post with a success notice.
+    # If the save fails, it renders the new template again, showing validation errors.
+
+    set_current_user
+    @post = @current_user.posts.build(post_params)
+    @post.comments_counter = 0
+    @post.likes_counter = 0
```
Comment (LionRouge1) [inline app/controllers/posts_controller.rb]: - [x] Kindly note that the class `PostController` is a child of the class `ApplicationController` and that mean it inherited all methods that is in the application_controller, Therefore we don't need to set the current user it have be already set in the `ApplicationController`. So I suggest that use remove the `set_current_user` and make use of the `current_user` method that you created in the `ApplicationController`

  Example:
```rb
@post = current_user.posts.build(post_params)
```

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -1,2 +1,37 @@
...
+
+  # The create action is responsible for handling the form submission to create a new comment.
+  # It first finds the user based on the user_id parameter,
+  # then finds the corresponding post based on the post_id parameter.
+  # It then initializes a new comment associated with the post and the user,
+  # using the comment_params method to permit only the necessary parameters (in this case, just :text).
+  def create
+    @user = User.find(params[:user_id]) # Get the user from params
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.new(comment_params.merge(user: @user))
```
Comment (LionRouge1) [inline app/controllers/comments_controller.rb]: - [x] Nice work creating the method to add new comment, everything works as it intend to :clap: However, according to the project requirement, comment should be create on the behalf of the user set in the `ApplicationController` (see image). Therefore I suggest you make use of the `current_user` method that you created in your `ApplicationController`. 
Screenshot from 2023-11-10 10-34-22
  **HIT:**  Change the line 18 of your code to look like this `@user = current_user`

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new, :create]do
+    resources :comments , only:[:index, :new, :create]
+    resources :likes , only:[:create]
+    end
   end
 
   resources :posts, only: [:index, :show]
```
Comment (LionRouge1) [inline config/routes.rb]: - [x] Kindly remove this resources (`resources :posts, only: [:index, :show]`) from your routes.rb file. It already defined in the block above.

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @Aqurds, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Create a method called `current_user` in the ApplicationController :heavy_check_mark: 
- Create forms to add new post and new comment :heavy_check_mark: 
- Create post and comment on the behalf of the user set in the ApplicationController :x: 

## Required Changes ♻️
- [x] According to the project requirement a user should be allow to like a post, but there is not a way for the user to like a post in your application. Kindly add a button that allow the user to add likes to a post.
Screenshot from 2023-11-10 10-49-40
 This comment is about the `app/views/posts/show.html.erb` file 

  **Example:**
```rb
    	<span class="float-end">
      	Comments: <%= @post.comments_counter %>, <%= button_to "likes: #{@post.likes.count}", user_post_likes_path(@post.author, @post) %>
    	</span>
``` 
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Aqurds) [reply app/controllers/posts_controller.rb]: Applied the suggested changes. Thanks ;)

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (Aqurds) [reply app/controllers/comments_controller.rb]: Updated according to suggestion :)

File: config/routes.rb (hunk 3 above)
Comment (Aqurds) [reply config/routes.rb]: Removed :)

**Review by Alfredbis29 — APPROVED** [review]
# STATUS: APPROVED   
 ## Hi  @Aqurds 
Your project is complete! There is nothing else to say other than... Great job clap Congratulations! :tada:
 
 TO HIGHLIGHT:
 
 * Your project is professional :white_check_mark:
 * Good Job :+1:
 
 Now you can merge this branch and move on 1st_place_medal: HAPPY CODING AND GOOD LUCK WITH THE COMING ONES :crossed_fingers: :


 
 ## Optional suggestions

- [x] N/A

 _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
 
Cheers and Happy coding! :clap:  :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me @Alfredbis29 in your question so I can receive the notification.
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**